### ISSUE TITLE
`:lua =` doesn't put quotes around strings

### ISSUE THREAD

**Issue by zeertzjq** [neovim/neovim#25379]
### Problem

`:lua =` doesn't put quotes around strings

### Steps to reproduce

1. Run `nvim --clean`
2. Run `:lua =''`
3. An empty message is printed

### Expected behavior

`""` is printed, which is the case in v0.7 and v0.8, and according to docs:
```
    Executes Lua chunk {chunk}. If {chunk} starts with "=" the rest of the
    chunk is evaluated as an expression and printed. `:lua =expr` or `:=expr` is
    equivalent to `:lua print(vim.inspect(expr))`.
```
and `:lua print(vim.inspect(''))` prints `""`.

### Neovim version (nvim -v)

v0.9.2, v0.10.0-dev-1223+gde93489789

### Vim (not Nvim) behaves the same?

N/A

### Operating system/version

Arch Linux

### Terminal name/version

kitty 0.29.2

### $TERM environment variable

xterm-kitty

### Installation

build from repo

**Comment by dundargoc** [comment]
673d2b52fa4335aa083c52e6686f0728e25b8ebd

**Comment by zeertzjq** [comment]
Actually this may make more sense, as it's consistent with `:echo`. But it doesn't match `:h lua`.

**Comment by zeertzjq** [comment]
Reopening: Not adding quotes makes it impossible to spot trailing whitespace in a string.

**Comment by zeertzjq** [comment]
Hmm, simply using `print(vim.inspect(%s))` won't work, as `vim.print()` can accept multiple arguments. And since it pretty-prints all values, it cannot accepts options. So this can only be solved by either also changing the behavior of `vim.print()` or adding a new (internal) function that adds quotes.

**Comment by justinmk** [comment]
> as `vim.print()` can accept multiple arguments

I think that feature should be dropped. It's just brain damage. And really restricts what we can do in the future with `vim.print`.

**Comment by zeertzjq** [comment]
Hmm, that means it will print only the first return value of a function returning multiple values.

**Comment by justinmk** [comment]
Callers can opt-in to that use-case via `vim.print({ foo() })`. It's a breaking change, but it's really not something we should have ever supported.

**Comment by lewis6991** [comment]
> Reopening: Not adding quotes makes it impossible to spot trailing whitespace in a string.

Adding quotes would make `vim.print` no longer appropriate as a printf-style debugging tool. Or at least makes it worse compared to `print` due to all the additional quotes that will be emitted.

General use case for me is of the format:

```
print(prefix, context1, context2, ..., item)
```

Where l:
- `prefix` is some static string I can search for.
- `context*` are context variables: bufnr, winid, time, etc
- `item` the thing I want to view

If all these things are a `string`, then `print` is exactly what I want.

**Comment by justinmk** [comment]
> Guess I'll just keep using `print`.

Yes. `vim.print` should be able to do more than just be a `print` alias.

**Comment by lewis6991** [comment]
From what's described, you basically just want:

```lua
vim.print = function(...)
  print(vim.inspect(...))
end
```

Which is just a print alias, and one we don't need.

There's been no proposal to make it do _more_, just a breaking change to make it _different_ without justification that solves any problem other than hypothetical future extensions.

**Comment by zeertzjq** [comment]
`vim.inspect()` does not support multiple arguments.

**Comment by zeertzjq** [comment]
The discussion is getting a bit out of hand. I'll probably just go with the second solution (private function).
